src/msw/toplevel.cpp

```diff
@@ -246,6 +249,34 @@ WXHWND wxTopLevelWindowMSW::MSWGetParent() const
     return (WXHWND)hwndParent;
 }
 
+bool wxTopLevelWindowMSW::HandleDPIChange(const wxSize newDPI, const wxRect newRect)
+{
+    if ( !m_perMonitorDPIaware )
+    {
+        return false;
+    }
+
+    wxSize oldDPI = m_activeDPI;
+
+    if ( oldDPI == wxDefaultSize )
+    {
+        oldDPI = GetDPI();
```

Comment:
Is this correct? If it is, i.e. if `GetDpiForWindow()` still returns the old DPI, I don't see why do we need to store `m_activeDPI` at all when we can just always retrieve it.

If it isn't, i.e. if `GetDpiForWindow()` already returns the new value here, then we should have initialized `m_activeDPI` before, i.e. it should never be `wxDefaultSize` here.